File: windows/Liferay.SDK/Service/V73/Syncdevice/SyncdeviceService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Syncdevice
{
	public class SyncdeviceService : ServiceBase
	{
		public SyncdeviceService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> RegisterSyncDeviceAsync(string type, long buildNumber, int featureSet, string uuid)
		{
			var _parameters = new JsonObject();

			_parameters.Add("type", type);
			_parameters.Add("buildNumber", buildNumber);
			_parameters.Add("featureSet", featureSet);
			_parameters.Add("uuid", uuid);

			var _command = new JsonObject()
			{
				{ "/sync.syncdevice/register-sync-device", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task UnregisterSyncDeviceAsync(string uuid)
		{
			var _parameters = new JsonObject();

			_parameters.Add("uuid", uuid);

			var _command = new JsonObject()
			{
				{ "/sync.syncdevice/unregister-sync-device", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}
	}
}

File: android/services-v72/src/main/java/com/liferay/mobile/android/v72/syncdevice/SyncdeviceService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v72.syncdevice;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class SyncdeviceService extends BaseService {

	public SyncdeviceService(Session session) {
		super(session);
	}

	public JSONObject registerSyncDevice(String type, long buildNumber, int featureSet, String uuid) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("buildNumber", buildNumber);
			_params.put("featureSet", featureSet);
			_params.put("type", checkNull(type));
			_params.put("uuid", checkNull(uuid));

			_command.put("/sync.syncdevice/register-sync-device", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void unregisterSyncDevice(String uuid) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("uuid", checkNull(uuid));

			_command.put("/sync.syncdevice/unregister-sync-device", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

}